src/3d/qgsrubberband3d.cpp

```diff
@@ -182,11 +182,19 @@ void QgsRubberBand3D::removePoint( int index )
 QgsRubberBand3D::~QgsRubberBand3D()
 {
   if ( mPolygonEntity )
-    delete mPolygonEntity;
+  {
+    mPolygonEntity.reset();
+  }
   if ( mLineEntity )
-    delete mLineEntity;
+  {
+    QObject::disconnect( mEngine, nullptr, mLineMaterial, nullptr );
```

Comment:
This shouldn't be required -- mLineMaterial is parented to mLineEntity, so will be deleted when it is. Same with mMarkerMaterial.